doc/userguide/devguide/internals/engines/alerts-flow.rst

```diff
@@ -0,0 +1,94 @@
+************
+Alerts' Flow
+************
+
+.. contents:: Table of Contents
+
+Important contents ::
+
+- rules prioritizing
+- Explain thresholding
+- Describe the flow
+- describe what each step does
+- describe output/product of each step
+
+Baseline
+========
+
+- `Suricata rules format <https://suricata.readthedocs.io/en/latest/rules/intro.html>`_
+
+Introduction
+============
+
+This document explains how the Suricata engine processes Alerts, focusing on 
+the main entities that interact and affect which alerts are actually kept for 
+any traffic that Suricata sees.
+
+We try to be accessible to newcomers, but some level of Suricata knowledge will
+be required for the big picture to make sense.
+
+Overall Concepts
+================
+
+In Suricata, an Alert represents that a rule was triggered by a certain traffic/
+packet. Internally, we represent said matches with a ``PacketAlert`` structure 
```

Comment:
random thought: would it make sense to link to https://doxygen.openinfosecfoundation.org/structPacketAlert__.html here?

Reply:
Makes sense! Gonna add it.

(on another note: Release the Kraken!!)